### PR TITLE
Add YearSortElement

### PR DESCRIPTION
**Pull request by mayhem**
AISOTT.

### CODE REVIEW

File: troi/sorts.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from operator import attrgetter
+
+import troi
+from troi import PipelineError, Recording
+
+
+class YearSortElement(troi.Element):
+    '''
+        Sort recordings by year.
```
Comment (alastair) [inline troi/sorts.py:9]: add documentation that says behaviour is undefined for multiple items with the same year

File: troi/sorts.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    @staticmethod
+    def outputs():
+        return [Recording]
+
+    def read(self, inputs):
+
+        def year_sorter(r):
+            if not r.year:
+                return 0
```
Comment (alastair) [inline troi/sorts.py:31]: This sorts items with no year first. Is this desired behaviour? My gut feeling is that they should be at the end.

File: troi/tests/test_sorts.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import unittest
+
+from troi import Artist, Recording
+import troi.sorts 
+
+
+class TestYearSortElement(unittest.TestCase):
+
+    def test_year_sort_element(self):
```
Comment (alastair) [inline troi/tests/test_sorts.py:9]: would be good to see a test with a None year.

**Review by alastair — CHANGES_REQUESTED** [review]

File: troi/sorts.py (hunk 2 above)
Comment (mayhem) [reply troi/sorts.py]: I didn't have a preference, so I changed it as you suggested.

File: troi/sorts.py (hunk 1 above)
Comment (mayhem) [reply troi/sorts.py:10]: Done.

**Review by alastair — APPROVED** [review]
looks good, thanks.

Eventually we might want to consider a sorting element that allows you to pass in your own sorting key, but that's not important yet.

**Pull request merged**